editor/src/components/editor/global-shortcuts.tsx

```diff
@@ -885,11 +893,26 @@ export function handleKeyDown(
         return []
       },
       [PASTE_TO_REPLACE]: () => {
-        return isSelectMode(editor.mode)
-          ? editor.selectedViews.map((target) => {
-              return EditorActions.pasteToReplace()
-            })
-          : []
+        if (isSelectMode(editor.mode)) {
+          const pasteToReplacePostActionMenuData: PasteToReplacePostActionMenuData = {
+            type: 'PASTE_TO_REPLACE',
+            targets: editor.selectedViews,
+            internalClipboard: editor.internalClipboard,
+          }
+
+          const defaultChoice = stripNulls([
+            PasteToReplaceWithPropsReplacedPostActionChoice(pasteToReplacePostActionMenuData),
+            PasteToReplaceWithPropsPreservedPostActionChoice(pasteToReplacePostActionMenuData),
+          ]).at(0)
+
+          if (defaultChoice != null) {
+            return [
+              EditorActions.startPostActionSession(pasteToReplacePostActionMenuData),
+              EditorActions.executePostActionMenuChoice(defaultChoice),
+            ]
+          }
+        }
+        return []
```

Comment:
should this be a shared function between this shortcut and the context menu? I created a similar shared function called `createWrapInGroupAction` because it was no longer just dispatching a single EditorAction

Reply:
good idea! fixed in [6043360](https://github.com/concrete-utopia/utopia/pull/3948/commits/604336085919131d4680d5884ff116844bab0dd7)